src/axom/primal/operators/clip.hpp

```diff
@@ -250,6 +279,53 @@ AXOM_HOST_DEVICE Polyhedron<T, 3> clip(const Tetrahedron<T, 3>& tet,
   return clip(hex, tet, eps, tryFixOrientation);
 }
 
+/*!
+ * \brief Clips a 3D hexahedron against a hexahedron in 3D, returning
+ *        the geometric intersection of the hexahedron and the hexahedron
+ *        as a polyhedron
+ *
+ *  This function clips the hexahedron by the 6 planes obtained from the
+ *  hexahedron's faces (normals point inward). Clipping the
```

Comment:
It seems that this function assumes that `hex2`'s faces are planar.  Perhaps also `hex1`.
This assumption should be explicitly noted

What happens if the faces are not planar?

Reply:
I changed this code so it will make a plane with 3 points of the hex face and check the other point to see whether it is sufficiently close to the plane. If not, then 2 planes per face are created. This should do a better job with non-planar hex faces. In the original code, I mainly adapted a pattern used elsewhere and I didn't think about non-planar faces.